src/librepgp/stream-sig.cpp

```diff
@@ -223,27 +221,26 @@ signature_get_keyid(const pgp_signature_t *sig, uint8_t *id)
 }
 
 bool
-signature_set_keyid(pgp_signature_t *sig, const uint8_t *id)
+signature_set_keyid(pgp_signature_t *sig, const pgp_key_id_t &id)
 {
-    pgp_sig_subpkt_t *subpkt;
-
-    if (!sig || !id) {
+    if (!sig) {
         return false;
     }
 
     if (sig->version < PGP_V4) {
-        memcpy(sig->signer, id, PGP_KEY_ID_SIZE);
+        sig->signer = id;
         return true;
     }
 
-    subpkt = signature_add_subpkt(sig, PGP_SIG_SUBPKT_ISSUER_KEY_ID, PGP_KEY_ID_SIZE, true);
+    pgp_sig_subpkt_t *subpkt =
+      signature_add_subpkt(sig, PGP_SIG_SUBPKT_ISSUER_KEY_ID, PGP_KEY_ID_SIZE, true);
     if (!subpkt) {
         return false;
     }
 
     subpkt->parsed = 1;
     subpkt->hashed = 0;
-    memcpy(subpkt->data, id, PGP_KEY_ID_SIZE);
+    memcpy(subpkt->data, id.data(), PGP_KEY_ID_SIZE);
```

Comment:
Can we add an assertion `id.size() == PGP_KEY_ID_SIZE`?

Reply:
@rrrooommmaaa Please see the updated PR.